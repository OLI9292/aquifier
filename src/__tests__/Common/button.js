import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import Button from '../../Components/Common/button';

const sizes = ['small', 'medium', 'large'];

describe('Button', () => {
  const testButton = size => {
    it(`should render a ${size} button correctly`, () => {
      const Button_ = Button[size];
      const output = shallow(<Button_ />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });    
  }  

  testButton('small')
  testButton('medium')
  testButton('large')
})