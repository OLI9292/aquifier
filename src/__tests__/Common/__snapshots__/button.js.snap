// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a large button correctly 1`] = `
<button
  className="sc-ifAKCX dRhFMX"
/>
`;

exports[`Button should render a medium button correctly 1`] = `
<button
  className="sc-EHOje hPWsyx"
/>
`;

exports[`Button should render a small button correctly 1`] = `
<button
  className="sc-bZQynM hAKfto"
/>
`;
